crates/server/src/store/blocklist.rs

```diff
@@ -268,7 +270,16 @@ impl BlocklistAuthority {
             .iter()
             .any(|entry| self.blocklist.contains_key(entry))
         {
-            info!("block list matched query {name}");
+            match client {
+                Some(info) if self.log_clients => {
+                    info!(
+                        "block list matched query {name} for query from client {}",
+                        info.src
+                    );
+                }
+                _ => info!("block list matched query {name}"),
```

Comment:
The pre-existing `info!()` log wasn't doing this, so feel free to ignore, but I think both the old/new log emissions would be nicer with the args structured instead of formatted into the message.